src/main/scala/org/zalando/spearheads/innkeeper/dao/InnkeeperSchema.scala

```diff
@@ -67,4 +73,35 @@ class InnkeeperPostgresSchema @Inject() (
       db.run(dropSchemaAction)
     }
   }
+
+  override def migrate(): Unit = {
+    val MAX_TRIES = 100
+    val url = config.getString("innkeeperdb.url")
+    val user = config.getString("innkeeperdb.user")
+    val password = config.getString("innkeeperdb.password")
+
+    val dataSource = new PGSimpleDataSource()
+    dataSource.setUrl(url)
+    dataSource.setUser(user)
+    dataSource.setPassword(password)
+
+    val flyway = new Flyway()
+    flyway.setDataSource(dataSource)
+
+    def migrateWithRetry(tries: Int): Unit = {
+      if (tries >= MAX_TRIES) {
+        throw new Exception("could not connect to the database for migration")
+      }
+
+      try {
+        flyway.migrate()
+      } catch {
+        case e: FlywayException =>
+          Thread.sleep(100L)
+          migrateWithRetry(tries + 1)
+      }
+    }
+
+    migrateWithRetry(0)
```

Comment:
I would give `MAX_TRIES` as a parameter to this function, and decrement the counter.